### ISSUE TITLE
[Bug] pulsar-admin 2.x links don't properly link to 3.2

### ISSUE THREAD

**Issue by illegalnumbers** [apache/pulsar#22828]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Read release policy

- [X] I understand that unsupported versions don't get bug fixes. I will attempt to reproduce the issue on a supported version of Pulsar client and Pulsar broker.

### Version

Chrome

### Minimal reproduce step

Visit https://pulsar.apache.org/docs/2.10.x/pulsar-admin/ and click the `We recommend you use the 3.2.x.` link

### What did you expect to see?

The 3.2 docs

### What did you see instead?

![Screenshot 2024-06-03 at 11 12 59 AM](https://github.com/apache/pulsar/assets/618650/c2bf5a5a-dcf0-432b-8b4d-091773fc6966)

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by lhotari** [comment]
@visortelle do you have a chance to check this one? thanks!

**Comment by visortelle** [comment]
@illegalnumbers this doc page was removed for newer versions.

You can find the `pulsar-admin` CLI reference here ➡️
https://pulsar.apache.org/reference/#/next/pulsar-admin/

---

2.10.x support period [ended a year ago](https://pulsar.apache.org/contribute/release-policy/) therefore I think we shouldn't handle it.

If it makes sense, we could make a HEAD request to the suggested URL to check if it is broken. And if the page doesn't exist, don't show the latest version suggestion link.

**Comment by illegalnumbers** [comment]
@visortelle I disagree with  your assessment of the issue. This page is still up and publicly available. Most search links to anything pulsar related links to 2.x versions or to older 3.x versions rather than `next`. In fact, the reason I used that in the first place is because it was the first link returned from Google.

It is myopic to consider that just because the support period has ended that the pages are removed and no one will find them. Additionally, this redirect should just work. Old versions of docs should link to new versions.

**Comment by visortelle** [comment]
> Most search links to anything pulsar related links to 2.x versions or to older 3.x versions rather than next.

There were attempts to fix it in the past. It improved the situation a bit, but not completely. We probably should make one more attempt to exclude old version pages from the Google index.

https://github.com/apache/pulsar-site/pull/506
https://github.com/apache/pulsar/issues/17898

> Additionally, this redirect should just work.

This page doesn't exist anymore.
Do you suggest handling each such case individually? I mean instead of simply removing pulsar-admin.md file, leave it and add a message that the page has been moved, or this page isn't relevant for newer Pulsar versions ~~(Say Hi! to Pulsar SQL)~~

We can introduce a way to override the version suggestion message and link on a per-page basis.

It could work together with this:

> If it makes sense, we could make a HEAD request to the suggested URL to check if it is broken. And if the page doesn't exist, don't show the latest version suggestion link.

Handling it properly looks like a task for one or two full days. @illegalnumbers your GitHub profile says that you work in SN. Maybe you're ready to review all the docs pages across all the versions? Unfortunately, I don't have time to do it for free.

**Comment by lhotari** [comment]
A quick hack could be to add some reasonable page as the redirect target.
Redirects could be made with Apache httpd syntax in https://github.com/apache/pulsar-site/blob/main/static/.htaccess

**Comment by lhotari** [comment]
Reopening since the "not found" page isn't nice. We can keep this open for contributions.
